### PR TITLE
Add defaults.go and related tests

### PR DESCRIPTION
**Pull request by robertkotcher**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/xeipuuv/gojsonschema
 
+go 1.14
```
Comment (svrana) [inline go.mod]: doesn't seem to match your commit message. why do you need to change it?

**Review by svrana — COMMENTED** [review]
be great to leave a real commit message

File: go.mod (hunk 1 above)
Comment (robertkotcher) [reply go.mod]: You must have commented as I was updating. This is now out of date

**Comment by robertkotcher** [comment]
> be great to leave a real commit message

@svrana If by "a real commit message" you are requesting for me to elaborate more on what exactly defaults.go does, please say something like that instead.

**Comment by svrana** [comment]
Did you get this code from somewhere? Please link to original patchset or somehow credit the original author if there is one.

**Pull request merged**